### ISSUE TITLE
`ln -s  /some_mount_path /etc/hosts` make mount driver disappear

### ISSUE THREAD

**Issue by eromoe** [microsoft/WSL#4334]
From this problem https://github.com/microsoft/WSL/issues/4300 .
I found the problem is hosts file .

Today, I reinstall ubuntu1804

1. can see file under /mnt/c/
1. set use default user to root
1. change `/etc/wsl.conf`
    ```
    # Enable extra metadata options by default
    [automount]
    enabled = true
    root = /
    options = "metadata,umask=22,fmask=11"
    mountFsTab = false

    # Enable DNS – even though these are turned on by default, we’ll specify here just to be explicit.
    [network]
    generateHosts = true
    generateResolvConf = true
    ```
1. `wsl.exe -t Ubuntu-18.04` and relaunch ubuntu1804 , can see files under /c/
1. use https://github.com/DDoSolitary/LxRunOffline  move ubuntu1804 to d:/ubuntu  (no matter have this step or not)
1. launch ubuntu shell, at this point still can see files under /c/

But after 

```
rm /etc/hosts
ln -s /c/Windows/System32/drivers/etc/hosts /etc/hosts
```

Can't see files under /c any more .

**Comment by eromoe** [comment]
Set `generateHosts = false`  seems solve this problem , but this should not be happen .

**Comment by benhillis** [comment]
Missing template including Windows version.  I was unable to reproduce this on the most recent Insiders build.

**Comment by benhillis** [comment]
Fixed in [18990](https://github.com/MicrosoftDocs/WSL/blob/live/WSL/release-notes.md#build-18980).

**Comment by WSLUser** [comment]
`ln -s /some_mount_path` is all that's needed to repro this. not specifically `/etc/hosts`.  I had used it to provide the user I  created after using `wsl --import` as part of setting it as the default user (by first changing the uid and gid and then using ln -s as part of changing perms from the old uid to the new uid (1000)). Of course the chmod/own failed on all /mnt/c paths but appears to be what caused the mount drives to no longer automount despite using /etc/fstab or simply the automount option in `wsl.conf`. Also I lost my permissions for my distro user to my ssh keys created with my Windows user (which did work with a distro I installed the appx manually )
